src/util/tubemap.js

```diff
@@ -841,31 +887,66 @@ function compareNoNodeReadsByPreviousY(a, b) {
   const segmentA = reads[a.readIndex].path[a.pathIndex];
   const segmentB = reads[b.readIndex].path[b.pathIndex];
   if (segmentA.order === segmentB.order) {
-    return a.previousY - b.previousY;
+    // we want to sort of the reverse when the segment is between a cycle
+    // this ensures a loop that starts on the outside, stays on the outside
+    // and loops are rolled in order
+    if (segmentA?.betweenCycle && segmentB?.betweenCycle) {
+      return b.previousY - a.previousY;
+    } else {
+      return a.previousY - b.previousY;
+    }
   }
   return segmentA.order - segmentB.order;
 }
 
 // compare read segments by where they are going to
 function compareReadOutgoingSegmentsByGoingTo(a, b) {
+  // Expect two arrays both containing 2 integers.
+  // The first index of each array contains the read index
+  // The second index of each array contains the path index
   let pathIndexA = a[1];
   let pathIndexB = b[1];
-  // let readA = reads[a[0]]
-  // let nodeIndexA = readA.path[pathIndexA].node;
-  let nodeA = nodes[reads[a[0]].path[pathIndexA].node];
-  let nodeB = nodes[reads[b[0]].path[pathIndexB].node];
+  let readIndexA = a[0];
+  let readIndexB = b[0];
```

Comment:
Can't we use argument destructuring here?
```suggestion
function compareReadOutgoingSegmentsByGoingTo([readIndexA, pathIndexA], [readIndexB, pathIndexB]) {
```